src/CaptureFile/FileFragmentInputStream.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FILE_FRAGMENT_INPUT_STREAM_H_
+#define FILE_FRAGMENT_INPUT_STREAM_H_
+
+#include <google/protobuf/io/zero_copy_stream.h>
+
+#include <optional>
+
+#include "OrbitBase/File.h"
+
+namespace orbit_capture_file_internal {
+
+// ZeroCopyInputStream implementation for a file fragment with offset and size.
+// this class is used to read protos from capture file sections and makes sure
+// we do not overread into other sections of the file.
+class FileFragmentInputStream : public google::protobuf::io::ZeroCopyInputStream {
+ public:
+  explicit FileFragmentInputStream(const orbit_base::unique_fd& fd, uint64_t file_offset,
+                                   uint64_t size, size_t block_size = 1 << 16)
+      : fd_{fd},
+        file_fragments_start_{file_offset},
+        file_fragments_end_{file_offset + size},
+        buffer_(block_size),
+        current_position_{file_offset} {
+    CHECK(size > 0);
+  }
+
+  bool Next(const void** data, int* size) override;
```

Comment:
Can you add a super brief comment to at least the first two methods?

Reply:
added brief comments and links

Reply:
Thanks!